packages/taquito/src/operations/operation-emitter.ts

```diff
@@ -81,17 +83,33 @@ export abstract class OperationEmitter {
     const publicKeyHash = pkh ? pkh : await this.signer.publicKeyHash();
     let counterPromise: Promise<string | undefined> = Promise.resolve(undefined);
 
+    // initializes a currentVotingPeriod if the operation is a ballot op
+    let currentVotingPeriodPromise: Promise<VotingPeriodBlockResult | undefined> =
+      Promise.resolve(undefined);
+    ops.find(async (op) => {
+      if (op.kind === 'ballot') {
+        try {
+          currentVotingPeriodPromise = this.rpc.getCurrentPeriod();
+        } catch (e) {
+          throw new RPCResponseError(
+            `Failed to get the current voting period index: ${JSON.stringify(e)}`
+          );
+        }
+      }
+    });
```

Comment:
what is the return of this.rpc.getCurrentPeriod() when there is no period? 
could this be simplified to `let currentVotingPeriod = op.find.... -> return this.rpc.getCurrentPeriod()? 

just was odd seeing the promise.resolve(undefined)

Reply:
a node will always have a voting period, but the undefined is a way to catch cases where, say, the node doesn't respond correctly.